docs/source/topics/events.rst

```diff
@@ -18,33 +18,90 @@ to events.
 Event Types
 -----------
 
-The table below shows all of the events emitted by botocore.  In some cases,
-the events are listed as ``<service>.<operations>.bar``, in which ``<service>``
-and ``<operation>`` are replaced with a specific service and operation, for
-example ``s3.ListObjects.bar``.
-
-.. list-table:: Events
-   :header-rows: 1
-
-   * - Event Name
-     - Occurance
-     - Arguments
-     - Return Value
-   * - **service-created**
-     - Whenever a service is created via the Sessions ``get_service``
-       method.
-     - ``service`` - The newly created :class:`botocore.service.Service`
-       object.
-     - Ignored.
-   * - **before-call.<service>.<operation>**
-     - When an operation is being called (``Operation.call``).
-     - ``operation`` - The newly created :class:`botocore.operation.Operation`
-       object.
-     - Ignored.
-   * - **after-call.<service>.<operation>**
-     - After an operation has been called, but before the response is parsed.
-     - ``http_response`` - The HTTP response, ``parsed`` - The parsed data.
-     - Ignored.
+The list below shows all of the events emitted by botocore.  In some cases, the
+events are listed as ``event-name.<service>.<operations>``, in which
+``<service>`` and ``<operation>`` are replaced with a specific service and
+operation, for example ``event-name.s3.ListObjects``.
+
+* ``'before-call.<service>.<operation>'``
+* ``'before-send.<service>.<operation>'``
+* ``'after-call.<service>.<operation>'``
+
+
+before-call
+~~~~~~~~~~~~~~~~~~~~~
+
+:Full Event Name:
+  ``'before-call.<service>.<operation>'``
+
+:Description:
+  This event is emitted when an operation is called and provides access to the
+  parameters that will be passed when calling the operation.
+
+:Keyword Arguments Emitted:
+
+  :type params: dict
+  :param params: A dictionary representing the keyword arguments that will be
+                 used to invoke the operation call.
+
+  :type context: dict
+  :param context: A dictionary representing the operation context. Additional
+                  information can be stored in this context to share state
+                  between events for the operation call.
+
+:Expected Return Value: A tuple of (http_response, parsed_response) where the
```

Comment:
We can return `None` here as well. We may want to follow the wording from `before-send` on how returning will affect the behavior.